### DISCUSSION TITLE
When the Base URL Is Not '/'

### DISCUSSION THREAD

**Discussion by mitchell-up** [TanStack/router#1242] [Ideas]
## Problem
We often encouner situations where it's necessary to change our project's base URL. This adjustment can be made by configuring the base URL in `vite.config.js`.

```js
export default defineConfig({
  base: '/your-base-url',
  plugins: [
    react({}),
    TanStackRouterVite({}),
  ],
})
```

However, after setting the base URL, we're encountering a 404 error when accessing the pages. This issue arises because the index route path '/' has been changed to '/your-base-url', but there aren't any correct matchings in `routeTree.gen.ts`.

```js
// Create Virtual Routes

const IndexLazyImport = createFileRoute('/')()

// Create/Update Routes

const IndexLazyRoute = IndexLazyImport.update({
  path: '/',
  getParentRoute: () => rootRoute,
} as any).lazy(() =>
  import('./routes//index.lazy').then(
    (d) => d.Route,
  ),
)
```

Of course, one way to solve this problem is by adding a directory named 'your-base-url' and moving the routes files into that folder.

## Idea
What if we consider setting the base configuration option in `TanStackRouterVite`? By specifying the base URL in `TanStackRouterVite`, it would automatically adapt '/' to '/your-base-url'.

I'm open to discussing this idea further or exploring other potential solutions you might have in mind.

**Comment by schiller-manuel** [comment]
did you try out https://tanstack.com/router/v1/docs/framework/react/api/router/RouterOptionsType#basepath ?

**Reply by mitchell-up** [reply]
The method you provided worked perfectly! The issue with the base URL has been resolved using this method :). I will close this discussion.
